scripts/setup.py

```diff
@@ -78,13 +78,62 @@ def download_and_install_package(url_to_retrieve, filename):
             downloaded.
         filename: string. The name of the tar file.
     """
+
     python_utils.url_retrieve(url_to_retrieve, filename=filename)
     tar = tarfile.open(name=filename)
     tar.extractall(path=common.OPPIA_TOOLS_DIR)
     tar.close()
     os.remove(filename)
 
 
+def download_and_install_node_on_windows(url_to_retrieve, filename):
+    """Downloads and installs node in Oppia tools directory for Windows.
+
+    Args:
+        url_to_retrieve: string. The url to retrieve node.
+        filename: string. The name of the tar file.
+    """
+    python_utils.url_retrieve(url_to_retrieve, filename=filename)
+    subprocess.check_call(
+        ['powershell.exe', '-c', 'expand-archive',
+         filename, '-DestinationPath',
+         common.OPPIA_TOOLS_DIR])
+
+
+def download_and_install_node():
+    """Download and install node to Oppia tools directory."""
+
+    if common.is_x64_architecture():
+        architecture = 'x64'
+    else:
+        architecture = 'x86'
+
+    outfile_name = 'node-download'
+
+    if common.is_windows_os():
+        extension = '.zip'
+        node_file_name = 'node-v%s-win-%s' % (
+            common.NODE_VERSION, architecture)
+        url_to_retrieve = 'https://nodejs.org/dist/v%s/%s%s' % (
+            common.NODE_VERSION, node_file_name, extension)
+        download_and_install_node_on_windows(url_to_retrieve, outfile_name)
```

Comment:
Actually I think you can just inline the logic here. It's a bit cleaner not to have distro-specific functions here (since it raises the question of "where's the corresponding function for distro X").

Reply:
Updated